### DISCUSSION TITLE
もっとコメントを重視したUIに変えてほしい

### DISCUSSION THREAD

**Discussion by JunichiIto** [increments/qiita-discussions#234] [要望・アイデア・不具合]
これは以前からぜひ改善してもらいたいと思っている内容です。

## 現状のUIの問題
現状のUIでは記事を表示してもかなり下までスクロールしないとコメントを確認できません。

以下は本文の終端からコメント欄までの距離を矢印で示した図です。

PCの場合
![qiita com_gen_signup_items_96f88d36061938d29f96](https://user-images.githubusercontent.com/1148320/170845261-f6c81d5e-b515-4871-8c57-c59182ce8428.png)

スマホの場合
![qiita com_gen_signup_items_96f88d36061938d29f96(iPhone SE)](https://user-images.githubusercontent.com/1148320/170845265-8118efad-8b50-4678-8056-073dac3aafac.png)

また、ホームやタイムラインの一覧にも各記事のコメント数は表示されていません。
<img width="796" alt="Screen Shot 2022-05-29 at 7 51 55" src="https://user-images.githubusercontent.com/1148320/170845297-f24a548d-3405-4a10-b509-66ff6042ed3c.png">

僕に言わせると、記事コメントは読者が意識して自分から読みに行かない限り、その存在に気づけないUIになっています。

## 記事コメントはなぜ重要なのか

が、コメントは記事本文に匹敵するぐらい重要な情報だと考えています。
というのも、記事本文の間違いをコメント欄で指摘している場合があるからです。

具体例を挙げると以下のようなケースです。
https://qiita.com/gen_signup/items/96f88d36061938d29f96

もちろん編集リクエストで修正を依頼するという手もありますが、編集リクエストは送ってもスルーされることがあります（古い記事ですでに執筆者がアクティブユーザーでない場合など）。
また、記事の内容が根本的に間違っている場合は記事全体を書き替える必要があるため、編集リクエストで対処するのは非現実的です。

プログラミング初心者さんの場合、

- エラーメッセージでググる
- 上位に表示されたQiita記事をクリックする
- 本文を読んでその解決策を自分のコードに適用する

といった短絡的な行動を取りがちです。
こうなると、いくらコメント欄で「おーい、この解決策は間違ってるよ！！」と叫んでも今のUIでは気付いてもらえません。
間違った解決策が初心者プログラマのコードに適用されるのも不幸ですし、場合によっては「このエラーはこれで直りました！」と間違った解決策をさらにQiitaに投稿して「間違った知識の拡大再生産」が行われることも珍しくありません。
こういった不幸の連鎖はQiita社の「エンジニアを最高に幸せにする」というミッションに反していると思います。

## その他、コメント欄が有益な理由

また、コメント欄は間違いの指摘だけでなく、「この記事は役に立ちました」とか「ちょっと異議あり！」といったfor or againstの議論が上がったり、「こういうアプローチもありますよ」という追加情報も載っていたりして、非常に有益であることが多いです。
なので、僕としては記事本文の内容だけでなく、コメント欄がどれくらい盛り上がっているのかという情報もぱっと認知したいです。

## 改善案
僕はUIデザインの専門家ではないので、具体的な改善方法はQiitaさんにお任せしますが、たとえば以下のようなUIになっていると嬉しいです。

### ファーストビューでコメント数を確認できるようにする
<img width="1418" alt="Screen Shot 2022-05-29 at 8 19 48" src="https://user-images.githubusercontent.com/1148320/170845865-17f02333-9f65-48fe-bdb7-27d56015922b.png">

### 本文の直後にコメント欄を移動する
<img width="1274" alt="Screen Shot 2022-05-29 at 8 22 52" src="https://user-images.githubusercontent.com/1148320/170845878-f419939a-717f-4f59-9d56-7d6a1a025c52.png">

もしくは本文の終わりにコメント数を表示させるとか。
（コメント欄の移動の方が気付いてもらいやすそうなのでベターですが）
<img width="1280" alt="Screen Shot 2022-05-29 at 8 21 04" src="https://user-images.githubusercontent.com/1148320/170845888-4f9e0b15-6c76-4b47-9e50-9bb63f0e6402.png">

### ホームやタイムラインの一覧にもコメント件数を表示する
コメント数を表示させることで「お、この記事はなんか盛り上がってるな」と気付きやすくなります。
<img width="796" alt="Screen Shot 2022-05-29 at 7 51 55 2" src="https://user-images.githubusercontent.com/1148320/170845916-27ffb888-9588-4621-94d3-b461deb5e435.png">

冒頭にも述べたとおり、この点は僕の中ではかなり重要度が高い要望なので、ぜひご検討いただけるとありがたいです。

**Comment by boc1211** [comment]
横からすみません。
コメント欄の有益さについては仰る通りだとは思うんですが、どちらかと言うと考えるべきはコメント欄の「有害さ」かなと思います。

昔のQiitaではコメント数とかが一目で分かるようなデザインだった記憶があるんですが、じゃあその時の雰囲気がどうだったかっていうと、個人的には「荒れやすいなぁ」というような印象でした。
コメント数が一目で分かると、「その記事を見たい人」ではなく、「盛り上がってる記事に横槍入れたい人」が集まりやすいのかなって個人的には思ってました。

そういう思い出もあるので、個人的にはかなり慎重に考えて欲しいかなとは思います。

**Reply by JunichiIto** [reply]
コメントありがとうございます。なるほど、そう言われればそういうケースもあるかもしれませんね。

ただ、個人的な感想を言えばコメント欄は有害だって思われる（デメリットがメリットを上回ってしまう）のはすごく寂しいものがあります。
具体的にどうすればいいのかは僕もいいアイデアはないですが、「害があるからコメント欄を引っ込める」のではなく、害を下げて有益さを上げるアプローチは何かないのかな〜と思ったりします。

**Comment by tanaka-asami415** [comment]
@JunichiIto @boc1211
ご意見ありがとうございます🙏

JunichiIto さん：画像などもつけていただき、ありがとうございます🙇

boc1211 さんのコメントでもいただいたようなことや
「この記事は最終更新日から5年以上が経過しています」といった古い記事の警告を見ずに
コメントに飛んでしまうのはどうか…など、色々な兼ね合いがあり、全部を叶えることは難しいかもしれませんが

記事本文の間違いの指摘など、折角コメントに記載してもらっているのに気づけないことは
非常にもったいないことですので、何かしらの方法で改善できるよう頑張りたいと思います！:muscle:

ただ、大分先になってしまいそうなので、気長にお待ちいただけるとありがたいです🙏💦

**Reply by JunichiIto** [reply]
返信ありがとうございます。

> 「この記事は最終更新日から5年以上が経過しています」といった古い記事の警告を見ずにコメントに飛んでしまうのはどうか

すいません、ここの意味がちょっとよくわかりませんでした。
古い記事であっても（警告が出ても出なくても）コメント欄は有益になりませんかね？
たとえば「この記事が書かれた当時とは異なり、今は○○というメソッドに置き換わってますよ」というようなコメントなら有益なコメントになると思います。

> 記事本文の間違いの指摘など、折角コメントに記載してもらっているのに気づけないことは
非常にもったいないことですので、何かしらの方法で改善できるよう頑張りたいと思います！💪

そうなんですよね。僕は記事の筆者以外にも読んでもらいたいコメントをよく書いているので、コメントなるべく目に付いた方がありがたいです。

> ただ、大分先になってしまいそうなので、気長にお待ちいただけるとありがたいです🙏💦

個人的には優先度が高いと思っているのですが、そこは仕方がないですね。
もうしばらく待つことにします。

**Reply by tanaka-asami415** [reply]
@JunichiIto 
失礼いたしました！前回勘違いをしておりました🙏💦

>「この記事は最終更新日から5年以上が経過しています」といった古い記事の警告を見ずにコメントに飛んでしまうのはどうか…

こちらは、ホームフィードから直接コメントに飛ばすことはせず
一旦、記事を見てもらった上で、コメント欄も見てほしいということだったのですが
そもそも、JunichiItoさんのご意見は、コメント数を表示させてはどうかということでした🙇大変失礼いたしました…💦

記事ページを開いた上でコメント欄に飛ばすことや、コメント数を目立たせることは
我々も重要だと思っているので、やっていきたいと思います！

>僕は記事の筆者以外にも読んでもらいたいコメントをよく書いているので、コメントなるべく目に付いた方がありがたいです。

本当にありがたいことです🙏 JunichiItoさん のようなユーザー様の行いが積み重なって、
Qiitaが「エンジニアにとって再利用性・汎用性の高い情報が集まる場」となっております！
引き続きQiitaをお願いいたします🙇🙇

**Reply by JunichiIto** [reply]
@tanaka-asami415 返信ありがとうございます。もろもろ了解しました！
こちらこそ引き続きよろしくお願いいたします。

**Comment by kimkim0814** [comment]
@JunichiIto
大変お待たせいたしました!

>もしくは本文の終わりにコメント数を表示させるとか。
>（コメント欄の移動の方が気付いてもらいやすそうなのでベターですが）

上記の意見を参考に、コメントへアクセスしやすくするために記事の下部にコメントへのページ内リンクを追加しました。
[記事ページの下部に各種ボタンを追加しました](https://qiita.com/release-notes/2024-04-08)

当初は、上部にコメントへのページ内リンクも付けることを考えていましたが
リンクカードなどの埋め込みコンテンツがある記事では、ジャンプ時に位置がズレてしまうため今回は下部のみに設置いたしました🙇

この度は貴重なご意見・アイデアありがとうございました!

![](https://github.com/increments/qiita-discussions/assets/54876760/77ae345b-92b4-4df7-a5f0-8600d13eda3d)

**Reply by JunichiIto** [reply]
@kimkim0814 
おお、ついに実装されたんですね！とても良いと思います 🎉 
ご対応どうもありがとうございます！！
